### ISSUE TITLE
connect socket with path

### ISSUE THREAD

**Issue by vikoms** [doghappy/socket.io-client-csharp#206]
hi, I'm currently developing a mobile application using xamarin forms, I'm trying to connect a socket to the server with a url like this:
**server url : https://subdomain.name.education/socket/**

and the code in my client is like this
<img width="607" alt="Screen Shot 2021-09-17 at 3 11 04 PM" src="https://user-images.githubusercontent.com/36475127/133748406-3c7889da-8994-41ac-b773-42cb62b675ef.png">

but i get an exception like this
<img width="827" alt="Screen Shot 2021-09-17 at 3 14 11 PM" src="https://user-images.githubusercontent.com/36475127/133748826-16df9dec-bbb5-444b-94b6-d9cdaa43900c.png">

is there a solution for my problem?
thank you in advance

note : sorry I can't tell you the exact url of my server

cc: @doghappy

**Comment by doghappy** [comment]
path: https://socket.io/docs/v4/client-api/#:~:text=Description-,path,-/socket.io

The url you gave, the namespace is "socket"

try this:

```cs
            var socket = new SocketIO("https://subdomain.name.education/socket/", new SocketIOOptions
            {
                  ...
            });
```

btw, today I optimized the support for xamarin. If you need to use these features, you need to use the latest code instead of nuget. (In the next few days, I will release)


---

Optimized the support for Xamarin: the library will always try to connect to the server, and an exception will be thrown when the connection fails. The library catch some exceptions, such as TimeoutException, WebSocketException, HttpRequestException and OperationCanceledException. When these exceptions are thrown, the reconnection will retry to connect. But in Xamarin, the exceptions thrown may not be these types.

For Xamarin.Android you should add the following code:

```cs
    public partial class MainPage: ContentPage
    {
        public MainPage()
        {
            InitializeComponent();
        }

        public SocketIO Socket {get;}
    }

    ...

    public class MainActivity: global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            ...
            var app = new App();
            var mainPage = app.MainPage as MainPage;
            mainPage.Socket.AddExpectedException(typeof(Java.Net.SocketException));
            mainPage.Socket.AddExpectedException(typeof(Java.Net.SocketTimeoutException));
            LoadApplication(app);
        }

        ...
    }
```

I haven’t debugged in macOS yet, so I don’t know the specific exception of Xamarin.iOS

**Comment by vikoms** [comment]
@doghappy  hello, thanks in advance for responding to my issue, but after I did what you suggested, I still got the same exception, is there any other settings from the client side??

**Comment by vikoms** [comment]
by the way i'm using version 2.2.4

**Comment by doghappy** [comment]
What I can be sure of is that this library should use the same url as the js client.

Please try another url:

```cs
            var socket = new SocketIO("https://subdomain.name.education/socket", new SocketIOOptions
//                                                                       ^^^
            {
// do not set Path
            });
```

**Comment by vikoms** [comment]
is there any other way ?? I think the connection failed because my server url is using https, because if I connect the socket to http it will work fine

**Comment by vikoms** [comment]
issue closed move to [#212 Socket connect to https](https://github.com/doghappy/socket.io-client-csharp/issues/212)

**Comment by vikoms** [comment]
issue closed move to [#212 Socket connect to https](https://github.com/doghappy/socket.io-client-csharp/issues/212)
